src/db/repository/UserRepository.ts

```diff
@@ -23,6 +23,18 @@ export default class UserRepository {
 		}
 	}
 
+	async getUserAndUpdateById(id: string, updateParams: Partial<User>) {
+		try {
+			const updatedUser = await this.userModel
+				.findOneAndUpdate({ _id: id }, { $set: updateParams }, { new: true })
+				.lean()
+				.exec();
+			return updatedUser;
```

Comment:
You can just return the update statement, no need to assign the result first

Reply:
Ok I think that will make the code more concise and aligned with existed codebase

Reply:
Also, what's your opinion about assigning return value to an easy to understand name variable, for further debug/logging/doing more stuff using that variable? I want to learn more about coding convention/style guide from an experienced dev like you, thanks pretty much